fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/impl/ResourceDAOImpl.java

```diff
@@ -678,7 +682,11 @@ public Resource insert(Resource resource, List<ExtractedParameterValue> paramete
         double dbCallDuration;
 
         try {
-            stmtString = getSearchByIdsSql(resourceType);
+            if (includeResourceData) {
+                stmtString = getSearchByIdsSql(resourceType);
```

Comment:
mostly tangential but a random thought I had:  should we log the query (maybe level finer?) before we actually run it?

Reply:
I guess it gets logged in both success (fine) and most failure cases (severe), so it might be fine as-is.

Reply:
Correct, the query is logged as it was before so not worth changing, I think.